### PR TITLE
Add Basic auth to options

### PR DESCRIPTION
**Pull request by uamana**
Adds a new key to options named auth. Its value is a table with two keys: user and pass. If all keys set, call of SetBasicAuth is performed on request.

### CODE REVIEW

File: gluahttp.go [hunk 1]
```diff
@@ -4,7 +4,6 @@ import (
 	"context"
 	"errors"
 	"fmt"
-	"github.com/yuin/gopher-lua"
```
Comment (erikdubbelboer) [inline gluahttp.go:7]: Why did you remove this? That doesn't make any sense.

File: README.md [hunk 2]
```diff
@@ -210,6 +210,7 @@ func main() {
 | form    | String | Deprecated. URL encoded request body. This will also set the `Content-Type` header to `application/x-www-form-urlencoded` |
 | headers | Table  | Additional headers to send with the request |
 | timeout | Number/String | Request timeout. Number of seconds or String such as "1h" |
+| auth    | Table  | Basic auth for request. Set user and pass keys. |
```
Comment (erikdubbelboer) [inline README.md:213]: Can you describe what this value should look like here?

Comment (erikdubbelboer) [reply README.md:213]: Maybe with an example.

**Review by erikdubbelboer — CHANGES_REQUESTED** [review]

File: gluahttp.go (hunk 1 above)
Comment (uamana) [reply gluahttp.go:7]: This is done by autoformating for Golang in VSCode, sorry.

File: README.md (hunk 2 above)
Comment (uamana) [reply README.md:213]: Hello.
This value should look like: `auth={user="user", pass="pass"}`
Full example: `response, error_message = http.request("GET", "http://example.com", {query="page=1", auth={"user="user", pass="pass"}})`

**Comment by erikdubbelboer** [comment]
Thanks!
